### PR TITLE
Expanded metrics support to include Servo, Spectator, and Atlas

### PR DESCRIPTION
**Pull request by jkschneider**
* For Java 8 users, Spectator is the recommended metrics library as Servo is now deprecated in favor of Spectator
* Servo support now needs to be specifically enabled via @EnableServo, even if servo-core is included on the classpath
* Added starter modules for both Spectator and Atlas support
* ServoMetricReader defines a convention for mapping tag-based metrics to hiearchical names for rendering by the Actuator /metrics endpoint

### CODE REVIEW

**Comment by spencergibb** [comment]
@jkschneider thanks! I'll start looking at it.

File: spring-cloud-netflix-core/src/main/resources/META-INF/spring.factories [hunk 1]
```diff
@@ -20,4 +19,10 @@ org.springframework.cloud.client.discovery.EnableDiscoveryClient=\
 org.springframework.cloud.netflix.eureka.EurekaDiscoveryClientConfiguration
 
 org.springframework.cloud.client.circuitbreaker.EnableCircuitBreaker=\
-org.springframework.cloud.netflix.hystrix.HystrixCircuitBreakerConfiguration
\ No newline at end of file
+org.springframework.cloud.netflix.hystrix.HystrixCircuitBreakerConfiguration
+
+org.springframework.cloud.netflix.metrics.servo.EnableServo=\
```
Comment (spencergibb) [inline spring-cloud-netflix-core/src/main/resources/META-INF/spring.factories]: Quick question? Do you want these to be auto-configured, or only enabled via `@Enable*` annotations. I'm close to merging this :-)

Comment (spencergibb) [inline spring-cloud-netflix-core/src/main/resources/META-INF/spring.factories]: Don't make a change, just let me know.

**Comment by aheusingfeld** [comment]
@spencergibb I didn't have the time to take a closer look, yet, but is this in line with the changes introduced to the spring boot metrics in their upcoming 1.3.0 release?

**Comment by spencergibb** [comment]
@aheusingfeld I don't know. @dsyer?

File: spring-cloud-netflix-core/src/main/resources/META-INF/spring.factories (hunk 1 above)
Comment (jkschneider) [inline spring-cloud-netflix-core/src/main/resources/META-INF/spring.factories]: I'm thinking enable via annotations for two reasons:
1.  This implementation is a bit more opinionated -- adds request metrics to MVC endpoints, etc.
2.  We need to be able to make a binary decision between Spectator and Servo.  I *think* that everyone on Java 8 would want Spectator, but I can't presume to understand every use case.

Comment (spencergibb) [inline spring-cloud-netflix-core/src/main/resources/META-INF/spring.factories]: Just to document from our slack conversation, I'm going to change `@Enable{Servo|Specator}` to auto configure with `@ConditionalOnMissingClass(Registry.class)` for Servo and `@ConditionalOnClass(Registry.class)` for Spectator.

**Comment by jkschneider** [comment]
Updated to remove the servo-atlas plugin to avoid a transitive dependency on spectator via rxnetty.  This functionality is now built directly into spring-cloud-netflix-core.

**Comment by jkschneider** [comment]
To test against a real Atlas instance, run this:

```bash
#!/usr/bin/env bash
mkdir -p dist
java -jar dist/atlas-1.4.4-standalone.jar dist/memory.conf
```

**Comment by spencergibb** [comment]
@jkschneider looking good.

I see metrics like this from `/env`:
```
{
    "mem": 146432,
    "mem.free": 92195,
    "processors": 8,
    "instance.uptime": 227002,
    "uptime": 231830,
    "systemload.average": 5.0166015625,
    "heap.committed": 146432,
    "heap.init": 262144,
    "heap.used": 54236,
    "heap": 3728384,
    "threads.peak": 22,
    "threads.daemon": 19,
    "threads.totalStarted": 28,
    "threads": 22,
    "classes": 8210,
    "classes.loaded": 8210,
    "classes.unloaded": 0,
    "gc.ps_scavenge.count": 28,
    "gc.ps_scavenge.time": 89,
    "gc.ps_marksweep.count": 4,
    "gc.ps_marksweep.time": 395,
    "httpsessions.max": -1,
    "httpsessions.active": 0,
    "datasource.primary.active": 0,
    "datasource.primary.usage": 0,
    "gauge.response.admin.env": 39,
    "gauge.response.star-star.favicon.ico": 24,
    "counter.status.200.star-star.favicon.ico": 1,
    "counter.status.200.admin.env": 1,
    "eurekaClient.registration.lastHeartbeatSec_00060(class=org.springframework.cloud.netflix.eureka.CloudEurekaClient,type=GAUGE)": 0,
    "timeouts(class=TimedSupervisorTask,id=heartbeat,type=COUNTER)": 0,
    "rejectedExecutions(class=TimedSupervisorTask,id=heartbeat,type=COUNTER)": 0,
    "throwables(class=TimedSupervisorTask,id=heartbeat,type=COUNTER)": 0,
    "threadPoolUsed(class=TimedSupervisorTask,id=heartbeat,type=GAUGE)": 0,
    "eurekaClient.registry.lastUpdateSec_00060(class=org.springframework.cloud.netflix.eureka.CloudEurekaClient,type=GAUGE)": 0,
    "eurekaClient.registry.lastUpdateSec_00030(class=org.springframework.cloud.netflix.eureka.CloudEurekaClient,type=GAUGE)": 0,
    "eurekaClient.registry.lastUpdateSec_00120(class=org.springframework.cloud.netflix.eureka.CloudEurekaClient,type=GAUGE)": 0,
    "eurekaClient.registration.lastHeartbeatSec_00030(class=org.springframework.cloud.netflix.eureka.CloudEurekaClient,type=GAUGE)": 0,
    "eurekaClient.registry.lastUpdateSec_00480(class=org.springframework.cloud.netflix.eureka.CloudEurekaClient,type=GAUGE)": 0,
    "DiscoveryClient_GetServiceUrlsFromDNS(class=CloudEurekaClient,statistic=totalTime,type=NORMALIZED,unit=MILLISECONDS)": 0,
    "DiscoveryClient_GetServiceUrlsFromDNS(class=CloudEurekaClient,statistic=count,type=NORMALIZED,unit=MILLISECONDS)": 0,
    "DiscoveryClient_GetServiceUrlsFromDNS(class=CloudEurekaClient,statistic=min,type=GAUGE,unit=MILLISECONDS)": 0,
    "DiscoveryClient_GetServiceUrlsFromDNS(class=CloudEurekaClient,statistic=max,type=GAUGE,unit=MILLISECONDS)": 0,
    "DiscoveryClient_Register(class=CloudEurekaClient,statistic=totalTime,type=NORMALIZED,unit=MILLISECONDS)": 6.130216666666667,
    "DiscoveryClient_Register(class=CloudEurekaClient,statistic=count,type=NORMALIZED,unit=MILLISECONDS)": 0.016666666666666666,
    "DiscoveryClient_Register(class=CloudEurekaClient,statistic=min,type=GAUGE,unit=MILLISECONDS)": 367.813,
    "DiscoveryClient_Register(class=CloudEurekaClient,statistic=max,type=GAUGE,unit=MILLISECONDS)": 367.813,
    "DiscoveryClient_Refresh(class=CloudEurekaClient,statistic=totalTime,type=NORMALIZED,unit=MILLISECONDS)": 0,
    "DiscoveryClient_Refresh(class=CloudEurekaClient,statistic=count,type=NORMALIZED,unit=MILLISECONDS)": 0,
    "DiscoveryClient_Refresh(class=CloudEurekaClient,statistic=min,type=GAUGE,unit=MILLISECONDS)": 0,
    "DiscoveryClient_Refresh(class=CloudEurekaClient,statistic=max,type=GAUGE,unit=MILLISECONDS)": 0,
    "DiscoveryClient_RefreshDelta(class=CloudEurekaClient,statistic=totalTime,type=NORMALIZED,unit=MILLISECONDS)": 0,
    "DiscoveryClient_RefreshDelta(class=CloudEurekaClient,statistic=count,type=NORMALIZED,unit=MILLISECONDS)": 0,
    "DiscoveryClient_RefreshDelta(class=CloudEurekaClient,statistic=min,type=GAUGE,unit=MILLISECONDS)": 0,
    "DiscoveryClient_RefreshDelta(class=CloudEurekaClient,statistic=max,type=GAUGE,unit=MILLISECONDS)": 0,
    "DiscoveryClient_ReconcileHashCodeMismatch(class=CloudEurekaClient,type=COUNTER)": 0,
    "DiscoveryClient_Renew(class=CloudEurekaClient,statistic=totalTime,type=NORMALIZED,unit=MILLISECONDS)": 0.19191666666666665,
    "DiscoveryClient_Renew(class=CloudEurekaClient,statistic=count,type=NORMALIZED,unit=MILLISECONDS)": 0.03333333333333333,
    "DiscoveryClient_Renew(class=CloudEurekaClient,statistic=min,type=GAUGE,unit=MILLISECONDS)": 5.614999999999999,
    "DiscoveryClient_Renew(class=CloudEurekaClient,statistic=max,type=GAUGE,unit=MILLISECONDS)": 5.8999999999999995,
    "DiscoveryClient_Cancel(class=CloudEurekaClient,statistic=totalTime,type=NORMALIZED,unit=MILLISECONDS)": 0,
    "DiscoveryClient_Cancel(class=CloudEurekaClient,statistic=count,type=NORMALIZED,unit=MILLISECONDS)": 0,
    "DiscoveryClient_Cancel(class=CloudEurekaClient,statistic=min,type=GAUGE,unit=MILLISECONDS)": 0,
    "DiscoveryClient_Cancel(class=CloudEurekaClient,statistic=max,type=GAUGE,unit=MILLISECONDS)": 0,
    "DiscoveryClient_FetchRegistry(class=CloudEurekaClient,statistic=totalTime,type=NORMALIZED,unit=MILLISECONDS)": 0,
    "DiscoveryClient_FetchRegistry(class=CloudEurekaClient,statistic=count,type=NORMALIZED,unit=MILLISECONDS)": 0,
    "DiscoveryClient_FetchRegistry(class=CloudEurekaClient,statistic=min,type=GAUGE,unit=MILLISECONDS)": 0,
    "DiscoveryClient_FetchRegistry(class=CloudEurekaClient,statistic=max,type=GAUGE,unit=MILLISECONDS)": 0,
    "DiscoveryClient_Retry(class=CloudEurekaClient,type=COUNTER)": 0,
    "DiscoveryClient_Failed(class=CloudEurekaClient,type=COUNTER)": 0,
    "DiscoveryClient_Reregister(class=CloudEurekaClient,type=COUNTER)": 0,
    "eurekaClient.registration.lastSuccessfulHeartbeatTimePeriod(class=CloudEurekaClient,level=INFO,type=GAUGE)": 17713,
    "eurekaClient.registry.lastSuccessfulRegistryFetchTimePeriod(class=CloudEurekaClient,level=INFO,type=GAUGE)": 0,
    "eurekaClient.registry.lastUpdateSec_00015(class=org.springframework.cloud.netflix.eureka.CloudEurekaClient,type=GAUGE)": 0,
    "eurekaClient.registration.lastHeartbeatSec_00015(class=org.springframework.cloud.netflix.eureka.CloudEurekaClient,type=GAUGE)": 1,
    "eurekaClient.registration.lastHeartbeatSec_00240(class=org.springframework.cloud.netflix.eureka.CloudEurekaClient,type=GAUGE)": 0,
    "eurekaClient.registration.lastHeartbeatSec_00480(class=org.springframework.cloud.netflix.eureka.CloudEurekaClient,type=GAUGE)": 0,
    "eurekaClient.registry.lastUpdateSec_00240(class=org.springframework.cloud.netflix.eureka.CloudEurekaClient,type=GAUGE)": 0,
    "DiscoveryClient-HTTPClient_Reuse(class=NamedConnectionPool,id=DiscoveryClient-HTTPClient,type=COUNTER)": 7,
    "DiscoveryClient-HTTPClient_CreateNew(class=NamedConnectionPool,id=DiscoveryClient-HTTPClient,type=COUNTER)": 1,
    "DiscoveryClient-HTTPClient_Request(class=NamedConnectionPool,id=DiscoveryClient-HTTPClient,type=COUNTER)": 8,
    "DiscoveryClient-HTTPClient_Release(class=NamedConnectionPool,id=DiscoveryClient-HTTPClient,type=COUNTER)": 8,
    "DiscoveryClient-HTTPClient_Delete(class=NamedConnectionPool,id=DiscoveryClient-HTTPClient,type=COUNTER)": 0,
    "DiscoveryClient-HTTPClient_RequestConnectionTimer(class=NamedConnectionPool,id=DiscoveryClient-HTTPClient,statistic=totalTime,type=NORMALIZED,unit=MILLISECONDS)": 0.0010999999999999998,
    "DiscoveryClient-HTTPClient_RequestConnectionTimer(class=NamedConnectionPool,id=DiscoveryClient-HTTPClient,statistic=count,type=NORMALIZED,unit=MILLISECONDS)": 0.03333333333333333,
    "DiscoveryClient-HTTPClient_RequestConnectionTimer(class=NamedConnectionPool,id=DiscoveryClient-HTTPClient,statistic=min,type=GAUGE,unit=MILLISECONDS)": 0.033,
    "DiscoveryClient-HTTPClient_RequestConnectionTimer(class=NamedConnectionPool,id=DiscoveryClient-HTTPClient,statistic=max,type=GAUGE,unit=MILLISECONDS)": 0.033,
    "DiscoveryClient-HTTPClient_CreateConnectionTimer(class=NamedConnectionPool,id=DiscoveryClient-HTTPClient,statistic=totalTime,type=NORMALIZED,unit=MILLISECONDS)": 0.06241666666666666,
    "DiscoveryClient-HTTPClient_CreateConnectionTimer(class=NamedConnectionPool,id=DiscoveryClient-HTTPClient,statistic=count,type=NORMALIZED,unit=MILLISECONDS)": 0.016666666666666666,
    "DiscoveryClient-HTTPClient_CreateConnectionTimer(class=NamedConnectionPool,id=DiscoveryClient-HTTPClient,statistic=min,type=GAUGE,unit=MILLISECONDS)": 3.7449999999999997,
    "DiscoveryClient-HTTPClient_CreateConnectionTimer(class=NamedConnectionPool,id=DiscoveryClient-HTTPClient,statistic=max,type=GAUGE,unit=MILLISECONDS)": 3.7449999999999997,
    "connectionCount(class=NamedConnectionPool,id=DiscoveryClient-HTTPClient,level=INFO,type=GAUGE)": 1,
    "eurekaClient.registration.lastHeartbeatSec_00120(class=org.springframework.cloud.netflix.eureka.CloudEurekaClient,type=GAUGE)": 0,
    "Eureka-Connection-Cleaner-Time(class=ConnectionCleanerTask,statistic=totalTime,type=NORMALIZED,unit=MILLISECONDS)": 0.0005666666666666666,
    "Eureka-Connection-Cleaner-Time(class=ConnectionCleanerTask,statistic=count,type=NORMALIZED,unit=MILLISECONDS)": 0.03333333333333333,
    "Eureka-Connection-Cleaner-Time(class=ConnectionCleanerTask,statistic=min,type=GAUGE,unit=MILLISECONDS)": 0.015,
    "Eureka-Connection-Cleaner-Time(class=ConnectionCleanerTask,statistic=max,type=GAUGE,unit=MILLISECONDS)": 0.019,
    "Eureka-Connection-Cleaner-Failure(class=ConnectionCleanerTask,type=COUNTER)": 0
}
```

**Comment by spencergibb** [comment]
@jkschneider The metrics used to look like the following:

From [DefaultServoMetricNaming](https://github.com/spring-cloud/spring-cloud-netflix/blob/master/spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/servo/DefaultServoMetricNaming.java). We had a few issue raised and fixed about this.

```
{
    "mem": 156160,
    "mem.free": 20069,
    "processors": 8,
    "instance.uptime": 174896,
    "uptime": 180388,
    "systemload.average": 2.42041015625,
    "heap.committed": 156160,
    "heap.init": 262144,
    "heap.used": 136090,
    "heap": 3728384,
    "threads.peak": 28,
    "threads.daemon": 25,
    "threads.totalStarted": 34,
    "threads": 28,
    "classes": 8094,
    "classes.loaded": 8094,
    "classes.unloaded": 0,
    "gc.ps_scavenge.count": 26,
    "gc.ps_scavenge.time": 106,
    "gc.ps_marksweep.count": 3,
    "gc.ps_marksweep.time": 279,
    "httpsessions.max": -1,
    "httpsessions.active": 0,
    "datasource.primary.active": 0,
    "datasource.primary.usage": 0,
    "gauge.normalized.servo.eureka-connection-cleaner-time.count": 0.03333333333333333,
    "gauge.servo.discoveryclient_refresh.min": 0,
    "gauge.normalized.servo.discoveryclient_getserviceurlsfromdns.count": 0,
    "gauge.servo.eurekaclient.registration.lastheartbeatsec_00030": 0,
    "gauge.servo.discoveryclient_refreshdelta.max": 0,
    "gauge.servo.discoveryclient_fetchregistry.min": 0,
    "gauge.normalized.servo.discoveryclient_cancel.count": 0,
    "gauge.normalized.servo.discoveryclient_register.count": 0,
    "gauge.normalized.servo.eureka-connection-cleaner-time.totaltime": 0.0005666666666666666,
    "gauge.counter.servo.discoveryclient_failed": 0,
    "gauge.normalized.servo.discoveryclient_getserviceurlsfromdns.totaltime": 0,
    "gauge.counter.servo.discoveryclient-httpclient_release": 6,
    "gauge.servo.discoveryclient-httpclient_createconnectiontimer.max": 0,
    "gauge.servo.eurekaclient.registry.lastupdatesec_00030": 0,
    "gauge.normalized.servo.discoveryclient_fetchregistry.totaltime": 0,
    "gauge.servo.discoveryclient_fetchregistry.max": 0,
    "gauge.servo.discoveryclient_cancel.min": 0,
    "gauge.servo.eurekaclient.registry.lastupdatesec_00120": 0,
    "gauge.servo.eurekaclient.registry.lastupdatesec_00240": 0,
    "gauge.servo.threadpoolused": 0,
    "gauge.counter.servo.timeouts": 0,
    "gauge.counter.servo.discoveryclient_retry": 0,
    "gauge.counter.servo.eureka-connection-cleaner-failure": 0,
    "gauge.counter.servo.rejectedexecutions": 0,
    "gauge.normalized.servo.discoveryclient-httpclient_requestconnectiontimer.count": 0.03333333333333333,
    "gauge.servo.eurekaclient.registration.lastheartbeatsec_00060": 0,
    "gauge.counter.servo.discoveryclient-httpclient_request": 6,
    "gauge.counter.servo.discoveryclient_reconcilehashcodemismatch": 0,
    "gauge.normalized.servo.discoveryclient_renew.count": 0.03333333333333333,
    "gauge.servo.discoveryclient_cancel.max": 0,
    "gauge.normalized.servo.discoveryclient_refresh.totaltime": 0,
    "gauge.servo.eurekaclient.registry.lastupdatesec_00480": 0,
    "gauge.servo.discoveryclient_getserviceurlsfromdns.max": 0,
    "gauge.normalized.servo.discoveryclient_renew.totaltime": 0.13046666666666668,
    "gauge.counter.servo.discoveryclient_reregister": 0,
    "gauge.normalized.servo.discoveryclient_refreshdelta.count": 0,
    "gauge.servo.eurekaclient.registration.lastsuccessfulheartbeattimeperiod": 25981,
    "gauge.servo.eurekaclient.registry.lastupdatesec_00015": 0,
    "gauge.counter.servo.discoveryclient-httpclient_reuse": 5,
    "gauge.counter.servo.discoveryclient-httpclient_createnew": 1,
    "gauge.counter.servo.discoveryclient-httpclient_delete": 0,
    "gauge.servo.connectioncount": 1,
    "gauge.servo.discoveryclient_renew.min": 3.447,
    "gauge.normalized.servo.discoveryclient-httpclient_createconnectiontimer.totaltime": 0,
    "gauge.servo.discoveryclient_getserviceurlsfromdns.min": 0,
    "gauge.normalized.servo.discoveryclient-httpclient_requestconnectiontimer.totaltime": 0.0006666666666666666,
    "gauge.normalized.servo.discoveryclient_refreshdelta.totaltime": 0,
    "gauge.normalized.servo.discoveryclient-httpclient_createconnectiontimer.count": 0,
    "gauge.servo.eurekaclient.registration.lastheartbeatsec_00015": 1,
    "gauge.normalized.servo.discoveryclient_cancel.totaltime": 0,
    "gauge.servo.discoveryclient_renew.max": 4.381,
    "gauge.servo.discoveryclient_register.min": 0,
    "gauge.normalized.servo.discoveryclient_register.totaltime": 0,
    "gauge.counter.servo.throwables": 0,
    "gauge.servo.discoveryclient_refreshdelta.min": 0,
    "gauge.servo.eurekaclient.registry.lastsuccessfulregistryfetchtimeperiod": 0,
    "gauge.servo.eurekaclient.registration.lastheartbeatsec_00240": 0,
    "gauge.servo.eurekaclient.registration.lastheartbeatsec_00480": 0,
    "gauge.servo.eureka-connection-cleaner-time.min": 0.015,
    "gauge.servo.discoveryclient-httpclient_requestconnectiontimer.max": 0.022,
    "gauge.servo.discoveryclient_register.max": 0,
    "gauge.servo.discoveryclient-httpclient_createconnectiontimer.min": 0,
    "gauge.servo.discoveryclient_refresh.max": 0,
    "gauge.servo.eurekaclient.registration.lastheartbeatsec_00120": 0,
    "gauge.normalized.servo.discoveryclient_fetchregistry.count": 0,
    "gauge.servo.eurekaclient.registry.lastupdatesec_00060": 0,
    "gauge.normalized.servo.discoveryclient_refresh.count": 0,
    "gauge.servo.eureka-connection-cleaner-time.max": 0.019,
    "gauge.servo.discoveryclient-httpclient_requestconnectiontimer.min": 0.018
}
```

**Comment by jkschneider** [comment]
@spencergibb is it necessary to preserve old style naming?

**Comment by spencergibb** [comment]
@dsyer said
> Yeah, we should try and re-capture the names

I'm don't know all of the reasoning. I just know we changed it before because of user submitted issues. Seems like we would it would seem like a regression, unless we call it a breaking change.

**Comment by dsyer** [comment]
I don't think we should make a breaking change really (at least not just to support one or two libraries from Netflix that not everyone will care about or use). Period-separated metric names are just a lowest common denominator for such things in all the tools we looked at. It's not as rich as other formats but it should be lossless if you stick to a naming convention of some sort.

**Comment by jkschneider** [comment]
As mentioned in #517, the existing naming mechanism was already flawed.  `DefaultServoMetricNaming`'s prepending of "servo." on the front of metric names (that may have already been prepended with "servo.") leads to a combinatorial explosion of metric names and a subsequent OOM.

At the time, I didn't think the solution was quite as simple as checking whether the metric name already has a "servo" prefix, but can't remember the details.

I mention this only to say that it will take some work to not only restore the old metric names but ensure that a mechanism that resembles the old naming strategy doesn't leak memory.

**Comment by jkschneider** [comment]
@dsyer I initially felt comfortable changing the format because I believed the intent was to replace Boot's hierarchical metric system with a tagged/dimensional one.

**Comment by jkschneider** [comment]
Having thought about this a bit, I'm going to provide both options (two implementations of ServoMetricNaming) and default to the old style.

**Comment by spencergibb** [comment]
@jkschneider That sounds like a great option!

**Comment by jkschneider** [comment]
Updated with both naming schemes.

**Comment by spencergibb** [comment]
Now seeing this as metrics:

```
{
    "mem": 156160,
    "mem.free": 102834,
    "processors": 8,
    "instance.uptime": 18295,
    "uptime": 23993,
    "systemload.average": 4.16357421875,
    "heap.committed": 156160,
    "heap.init": 262144,
    "heap.used": 53325,
    "heap": 3728384,
    "threads.peak": 27,
    "threads.daemon": 25,
    "threads.totalStarted": 33,
    "threads": 27,
    "classes": 8477,
    "classes.loaded": 8477,
    "classes.unloaded": 0,
    "gc.ps_scavenge.count": 28,
    "gc.ps_scavenge.time": 107,
    "gc.ps_marksweep.count": 4,
    "gc.ps_marksweep.time": 492,
    "httpsessions.max": -1,
    "httpsessions.active": 0,
    "datasource.primary.active": 0,
    "datasource.primary.usage": 0,
    "gauge.response.admin.metrics": 5,
    "gauge.response.star-star.favicon.ico": 3,
    "counter.status.200.star-star.favicon.ico": 6,
    "counter.status.200.admin.metrics": 7,
    "gauge.servo.eurekaclient.registry.lastupdatesec_00060": 0,
    "gauge.servo.eurekaclient.registration.lastheartbeatsec_00060": 0,
    "counter.servo.timeouts": 0,
    "counter.servo.rejectedexecutions": 0,
    "counter.servo.throwables": 0,
    "gauge.servo.threadpoolused": 0,
    "normalized.servo.discoveryclient_getserviceurlsfromdns.totaltime": 0,
    "normalized.servo.discoveryclient_getserviceurlsfromdns.count": 0,
    "gauge.servo.discoveryclient_getserviceurlsfromdns.min": 0,
    "gauge.servo.discoveryclient_getserviceurlsfromdns.max": 0,
    "normalized.servo.discoveryclient_register.totaltime": 0,
    "normalized.servo.discoveryclient_register.count": 0,
    "gauge.servo.discoveryclient_register.min": 0,
    "gauge.servo.discoveryclient_register.max": 0,
    "normalized.servo.discoveryclient_refresh.totaltime": 0,
    "normalized.servo.discoveryclient_refresh.count": 0,
    "gauge.servo.discoveryclient_refresh.min": 0,
    "gauge.servo.discoveryclient_refresh.max": 0,
    "normalized.servo.discoveryclient_refreshdelta.totaltime": 0,
    "normalized.servo.discoveryclient_refreshdelta.count": 0,
    "gauge.servo.discoveryclient_refreshdelta.min": 0,
    "gauge.servo.discoveryclient_refreshdelta.max": 0,
    "counter.servo.discoveryclient_reconcilehashcodemismatch": 0,
    "normalized.servo.discoveryclient_renew.totaltime": 0,
    "normalized.servo.discoveryclient_renew.count": 0,
    "gauge.servo.discoveryclient_renew.min": 0,
    "gauge.servo.discoveryclient_renew.max": 0,
    "normalized.servo.discoveryclient_cancel.totaltime": 0,
    "normalized.servo.discoveryclient_cancel.count": 0,
    "gauge.servo.discoveryclient_cancel.min": 0,
    "gauge.servo.discoveryclient_cancel.max": 0,
    "normalized.servo.discoveryclient_fetchregistry.totaltime": 0,
    "normalized.servo.discoveryclient_fetchregistry.count": 0,
    "gauge.servo.discoveryclient_fetchregistry.min": 0,
    "gauge.servo.discoveryclient_fetchregistry.max": 0,
    "counter.servo.discoveryclient_retry": 0,
    "counter.servo.discoveryclient_failed": 0,
    "counter.servo.discoveryclient_reregister": 0,
    "gauge.servo.eurekaclient.registry.lastsuccessfulregistryfetchtimeperiod": 0,
    "gauge.servo.eurekaclient.registration.lastsuccessfulheartbeattimeperiod": 0,
    "gauge.servo.eurekaclient.registration.lastheartbeatsec_00030": 0,
    "gauge.servo.eurekaclient.registration.lastheartbeatsec_00240": 0,
    "normalized.servo.eureka-connection-cleaner-time.totaltime": 0,
    "normalized.servo.eureka-connection-cleaner-time.count": 0,
    "gauge.servo.eureka-connection-cleaner-time.min": 0,
    "gauge.servo.eureka-connection-cleaner-time.max": 0,
    "counter.servo.eureka-connection-cleaner-failure": 0,
    "gauge.servo.eurekaclient.registration.lastheartbeatsec_00480": 0,
    "gauge.servo.eurekaclient.registry.lastupdatesec_00120": 0,
    "gauge.servo.eurekaclient.registration.lastheartbeatsec_00015": 0,
    "counter.servo.discoveryclient-httpclient_reuse": 0,
    "counter.servo.discoveryclient-httpclient_createnew": 1,
    "counter.servo.discoveryclient-httpclient_request": 1,
    "counter.servo.discoveryclient-httpclient_release": 1,
    "counter.servo.discoveryclient-httpclient_delete": 0,
    "normalized.servo.discoveryclient-httpclient_requestconnectiontimer.totaltime": 0,
    "normalized.servo.discoveryclient-httpclient_requestconnectiontimer.count": 0,
    "gauge.servo.discoveryclient-httpclient_requestconnectiontimer.min": 0,
    "gauge.servo.discoveryclient-httpclient_requestconnectiontimer.max": 0,
    "normalized.servo.discoveryclient-httpclient_createconnectiontimer.totaltime": 0,
    "normalized.servo.discoveryclient-httpclient_createconnectiontimer.count": 0,
    "gauge.servo.discoveryclient-httpclient_createconnectiontimer.min": 0,
    "gauge.servo.discoveryclient-httpclient_createconnectiontimer.max": 0,
    "gauge.servo.connectioncount": 1,
    "gauge.servo.eurekaclient.registry.lastupdatesec_00015": 0,
    "gauge.servo.eurekaclient.registry.lastupdatesec_00480": 0,
    "gauge.servo.eurekaclient.registration.lastheartbeatsec_00120": 0,
    "gauge.servo.eurekaclient.registry.lastupdatesec_00240": 0,
    "gauge.servo.eurekaclient.registry.lastupdatesec_00030": 0
}
```

**Comment by spencergibb** [comment]
I think I'm going to merge it now and we can address anything with new PR's.

**Comment by spencergibb** [comment]
Merge with a bit of polish via b117030a47b136c326ef4be4b6a93831003a21ee

**Pull request closed without merging**